src/modules/image/imageApi.ts

```diff
@@ -12,74 +12,125 @@ import {
   ISearchResultV3DTO,
   ITagsSearchResultDTO,
   ISearchParamsDTO,
+  INewImageMetaInformationV2DTO,
+  openapi,
 } from "@ndla/types-backend/image-api";
 import { StringSort } from "../../containers/SearchPage/components/form/SearchForm";
-import {
-  resolveJsonOrRejectWithError,
-  apiResourceUrl,
-  fetchAuthorized,
-  throwErrorPayload,
-} from "../../util/apiHelpers";
-import { resolveJsonOrVoidOrRejectWithError } from "../../util/resolveJsonOrRejectWithError";
+import { throwErrorPayload, createAuthClient } from "../../util/apiHelpers";
+import { resolveJsonOATS } from "../../util/resolveJsonOrRejectWithError";
 import { createFormData } from "../../util/formDataHelper";
 
-const baseUrl = apiResourceUrl("/image-api/v3/images");
+const client = createAuthClient<openapi.paths>();
+
+export const postImage = async (
+  metadata: INewImageMetaInformationV2DTO,
+  file: Blob,
+): Promise<IImageMetaInformationV3DTO> => {
+  const res = await client.POST("/image-api/v3/images", {
+    body: {
+      metadata,
+      file,
+    },
+    bodySerializer(body) {
+      return createFormData(body.file, body.metadata);
+    },
+  });
 
-export const postImage = (formData: FormData): Promise<IImageMetaInformationV3DTO> =>
-  fetchAuthorized(`${baseUrl}`, {
-    method: "POST",
-    headers: { "Content-Type": undefined }, // Without this we're missing a boundary: https://stackoverflow.com/questions/39280438/fetch-missing-boundary-in-multipart-form-data-post
-    body: formData,
-  }).then((r) => resolveJsonOrRejectWithError<IImageMetaInformationV3DTO>(r));
+  return resolveJsonOATS(res);
+};
 
 export const fetchImage = (id: number | string, language?: string): Promise<IImageMetaInformationV3DTO> =>
-  fetchAuthorized(`${baseUrl}/${id}?language=${language}`).then((r) =>
-    resolveJsonOrRejectWithError<IImageMetaInformationV3DTO>(r),
-  );
+  client
+    .GET("/image-api/v3/images/{image_id}", {
+      params: {
+        path: {
+          image_id: typeof id === "string" ? Number(id) : id,
```

Comment:
Hvorfor er dette nødvendig? Blir openapi-ts sinna på oss dersom vi ikke caster? Isåfall, burde vi kanskje heller snevre inn hva vi tar inn som parameter-type?

Reply:
det er nødvendig siden backenden tar `Long` som type og frontenden av en eller annen grunn har lagret den som `string` på et tidspunkt.